docs/reference/content/upgrading.md

```diff
@@ -7,6 +7,24 @@ title = "Upgrade Considerations"
   pre = "<i class='fa fa-level-up'></i>"
 +++
 
+## Upgrading from 3.11.x
+
+In the upcoming 4.0 release, all deprecated API elements except those documented as "not currently scheduled for removal" will be removed.
+Currently the only deprecated API elements _not_ scheduled for removal are:
+
+* [`Mongo.getDB`]({{<apiref "com/mongodb/Mongo.html#getDB-java.lang.String-">}})
+* [`JsonMode.STRICT`]({{<apiref "org/bson/json/JsonMode.html#STRICT">}})
+
+To prepare for the 4.0 release, please compile with deprecation warnings enabled and replace all usage of deprecated API elements with their
+recommended replacements.
+
+Also, note that the 3.12 release (which includes support for MongoDB 4.2) will be the last release that is compatible with *Java 6
```

Comment:
Perhaps put this outside in a note above the upgrading from 3.11.x - just in case! ;)

eg:

```
{{% note class="important" %}}
The 3.x series of releases (which includes support to MongoDB 4.2) will be the last releases that are compatible with *Java 6 or Java 7*.  

The 4.0 Java driver will require a minimum of Java 8. Please note that support for MongoDB 4.4 will only be made available via a 4.x driver release.
{{% /note %}}
```

Reply:
Done.  Also changed the working a bit to hedge our bets, since I have a feeling we will get pressure to support 4.4 in some form in 3.x (even if it's just continuing to test against it but not supporting new features)